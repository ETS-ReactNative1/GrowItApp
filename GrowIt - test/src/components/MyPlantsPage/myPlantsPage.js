import React, { useState } from 'react';
import PropTypes from 'prop-types';
import { View, TouchableOpacity } from 'react-native';
import { ProgressBar, Colors } from 'react-native-paper';
import { CountDown } from 'react-native-countdown-component';
import { useSelector, useDispatch } from 'react-redux';
import { irrigatePlantAndUpdate } from '../../redux/actions/plantActions';
import { List, ListItem, Left, Body, Right, Thumbnail, Text } from 'native-base';
import Icon from 'react-native-vector-icons/MaterialCommunityIcons';
import ModalComponent from '../Modal/Modal';
import styles from './style';

const myPlantsPage = ({ navigation }) => {
  const dispatch = useDispatch();
  const myPlants = useSelector((state) => state.plantsReducer.myPlants);
  const [modalVisible, setModalVisible] = useState({ flag: false, obj: {} });

  return (
    <View>
      <ModalComponent modalVisible={modalVisible} setModalVisible={setModalVisible} />
      <List
        dataArray={myPlants}
        renderRow={(item) => (
          <ListItem avatar>
            <Left>
              <Thumbnail source={{ uri: item.imgUrl }} />
            </Left>

            <Body>
              <TouchableOpacity
                onPress={() => {
                  navigation.navigate('PlantPage', {
                    plantObj: { ...item },
                    buttonFlag: false
                  });
                }}
              >
                <Text style={styles.font}>{item.name}</Text>
                <Text note style={styles.font}>
                  Click for info
                </Text>
                <Text style={styles.font}>Irrigation status</Text>
                <ProgressBar
                  progress={item.nextIrrigate ? processPart(item) : 0}
                  color={Colors.blue800}
                />
                {item.nextIrrigate ? <Text style={styles.font}>Next irrigating</Text> : null}
                {item.nextIrrigate ? (
                  <ClockComponent countDownInSec={item.nextIrrigate - Date.now() / 1000} />
                ) : null}
              </TouchableOpacity>
            </Body>

            <Right>
              <Text note style={styles.font}>
                Added at {item.addedAt}
              </Text>
              <View style={styles.buttons}>
                <TouchableOpacity
                  onPress={() => {
                    setModalVisible({ flag: true, obj: item });
                  }}
                >
                  <Icon name="delete" size={25} color={Colors.red800} />
                </TouchableOpacity>
                <TouchableOpacity
                  onPress={() => {
                    dispatch(irrigatePlantAndUpdate(item));
                  }}
                  style={styles.irrigateBtn}
                >
                  <Icon name="water-pump" color={Colors.blue800} size={25} />
                </TouchableOpacity>
              </View>
            </Right>
          </ListItem>
        )}
        keyExtractor={(item) => item._id}
        ListEmptyComponent={
          <Text style={[styles.font, styles.emptyList]}>You dont have plants yet ü§∑‚Äç‚ôÇÔ∏è</Text>
        }
      />
    </View>
  );
};

const ClockComponent = ({ countDownInSec }) => {
  return (
    <CountDown
      size={11}
      style={styles.clock}
      until={countDownInSec}
      digitStyle={styles.clockDigitStyle}
      digitTxtStyle={styles.clockDigitTxtStyle}
      timeLabelStyle={styles.clockTimeLabelStyle}
      separatorStyle={styles.clockSeparatorStyle}
      timeToShow={['D', 'H', 'M', 'S']}
      timeLabels={{ d: 'D', h: 'H', m: 'M', s: 'S' }}
    />
  );
};

const processPart = (item) => {
  return (Date.now() / 1000 - (item.nextIrrigate - item.waterAmount)) / item.waterAmount;
};

ClockComponent.propTypes = {
  countDownInSec: PropTypes.number
};

export default myPlantsPage;
