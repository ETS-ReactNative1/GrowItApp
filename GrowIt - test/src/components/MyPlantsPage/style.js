import { StyleSheet } from 'react-native';

const styles = StyleSheet.create({
  font: {
    fontFamily: 'Comfortaa_600SemiBold'
  },
  emptyList: {
    fontSize: 22,
    textAlign: 'center',
    paddingTop: '20%'
  },
  buttons: {
    marginTop: 5,
    marginRight: 2,
    flex: 1,
    justifyContent: 'space-between'
  },
  irrigateBtn: {
    marginBottom: 15
  }
});

export default styles;
