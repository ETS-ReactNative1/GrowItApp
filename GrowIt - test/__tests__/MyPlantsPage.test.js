/**
 * @jest-environment jsdom
 */

import React from 'react';
import configureMockStore from 'redux-mock-store'
import thunk from 'redux-thunk'
import { render } from '@testing-library/react-native'
import MyPlantsPage from '../src/components/MyPlantsPage/MyPlantsPage'
import { Provider } from 'react-redux'
import { act } from "react-dom/test-utils";



const initialState = {
    plantsReducer: {
        location: {
            lat: 30, lon: 30
        },
        plantsList: [],
        types: [],
        myPlants:PlantsListMock

    }
}
const middlewares = [thunk]
const mockStore = configureMockStore(middlewares)
const store = mockStore(initialState);
describe('MyPlantsPage component Test', function () {
    beforeEach(() => { 
        store.clearActions();
    });
    let temp;
    test('render my plants page component snapshot ', async function () {
        await act( () => {
            const { baseElement } = render(<Provider store={store}><MyPlantsPage /></Provider>)
            temp = baseElement
          });

          expect(temp).toMatchSnapshot()
        
       
        
    });
    

});
const PlantsListMock = [
    {
      months: [7, 1, 8, 9, 2, 3, 4],
      _id: '5f0da3112f105dcd2e8b681b',
      name: 'Onion',
      family: 'Liliaceae',
      description:
        'The onion is a vegetable that is the most widely cultivated species of the genus Allium. Its close relatives include the garlic, scallion, shallot, leek, chive, and Chinese onion. ',
      type: 'vegetables',
      waterAmount: '43200',
      imgUrl: 'https://flevotrade.nl/wp-content/uploads/2019/11/flevotrade_onions_product.jpg',
      howToITreat:
        'Keep soil moist, water well but avoid spilling water and flooding the plant. Place the seedling near a window exposed to direct sunlight.',
      recommendedTemp: 30,
      recommendedHumidity: 30,
      recommendedClouds: 10
    },
    {
      months: [7, 1, 8, 9, 2, 3, 4],
      _id: '5f0da4922f105dcd2e8b681c',
      name: 'Cucumber ',
      family: 'Cucurbitaceae',
      description: `Its elongated fruit is juicy and very rich in water. Its green color is eaten raw, pickled in salt or pickled.The plant originated in India, where humans began growing it about 3, 000 years ago.Today, cucumber is a very common vegetable, and it is grown in greenhouses or open fields.`,
      type: 'vegetables',
      waterAmount: '259200',
      imgUrl: 'https://img.wcdn.co.il/f_auto,w_1200,t_54/1/4/9/8/1498895-46.jpg',
      howToITreat:
        'Water the cucumbers until the soil is thoroughly absorbed and maintain this moisture at all times.',
      recommendedTemp: 25,
      recommendedHumidity: 25,
      recommendedClouds: 15
    }
  ];