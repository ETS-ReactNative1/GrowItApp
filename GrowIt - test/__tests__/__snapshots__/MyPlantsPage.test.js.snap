// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyPlantsPage component Test render my plants page component snapshot  1`] = `
<View
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    collapsable={true}
    pointerEvents="box-none"
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View>
      <View
        style={
          Object {
            "alignItems": "center",
            "flex": 1,
            "justifyContent": "center",
          }
        }
      >
        <Modal
          animationType="slide"
          hardwareAccelerated={false}
          transparent={true}
          visible={false}
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "flex": 1,
                "justifyContent": "center",
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "white",
                  "borderColor": "black",
                  "borderRadius": 20,
                  "borderWidth": 1,
                  "elevation": 5,
                  "margin": 20,
                  "padding": 35,
                  "shadowColor": "#000",
                  "shadowOffset": Object {
                    "height": 2,
                    "width": 0,
                  },
                  "shadowOpacity": 0.25,
                  "shadowRadius": 3.84,
                }
              }
            >
              <Text
                style={
                  Object {
                    "fontFamily": "Comfortaa_600SemiBold",
                    "marginBottom": 15,
                    "textAlign": "center",
                  }
                }
              >
                Are you sure you want to remove 
                 from your garden?
              </Text>
              <View
                style={
                  Object {
                    "flexDirection": "row",
                    "marginTop": 5,
                  }
                }
              >
                <TouchableHighlight
                  activeOpacity={0.85}
                  delayPressOut={100}
                  underlayColor="black"
                >
                  <Text
                    style={
                      Array [
                        Object {
                          "fontFamily": "Comfortaa_600SemiBold",
                          "marginBottom": 15,
                          "textAlign": "center",
                        },
                        Object {
                          "color": "#2e7d32",
                        },
                      ]
                    }
                  >
                    Yes
                  </Text>
                </TouchableHighlight>
                <TouchableHighlight
                  activeOpacity={0.85}
                  delayPressOut={100}
                  underlayColor="black"
                >
                  <Text
                    style={
                      Array [
                        Object {
                          "fontFamily": "Comfortaa_600SemiBold",
                          "marginBottom": 15,
                          "textAlign": "center",
                        },
                        Object {
                          "color": "#c62828",
                          "marginLeft": 100,
                        },
                      ]
                    }
                  >
                    No
                  </Text>
                </TouchableHighlight>
              </View>
            </View>
          </View>
        </Modal>
      </View>
      <View
        ListEmptyComponent={
          <Styled(Text)
            style={
              Array [
                Object {
                  "fontFamily": "Comfortaa_600SemiBold",
                },
                Object {
                  "fontSize": 22,
                  "paddingTop": "20%",
                  "textAlign": "center",
                },
              ]
            }
          >
            You dont have plants yet ü§∑‚Äç‚ôÇÔ∏è
          </Styled(Text)>
        }
        style={Object {}}
      />
    </View>
  </View>
</View>
`;
